Sources/BraintreeCore/Analytics/BTAnalyticsService.swift

```diff
@@ -22,11 +22,14 @@ class BTAnalyticsService: Equatable {
     private let timerInterval = 15
 
     private static var timer: DispatchSourceTimer?
-    
+
+    private let configuration: BTConfiguration
+
     // MARK: - Initializer
 
-    init(apiClient: BTAPIClient) {
+    init(apiClient: BTAPIClient, configuration: BTConfiguration) {
```

Comment:
👏 I like this a lot!

Reply:
Nice, SOLID principle 💯 .

I was thinking about what would happen when the configuration expires, although the injection of the BTConfiguration solves a problem, we need to verify what would happen in this scenario where Configuration cache expires

Reply:
That's a good point. I do think we want to avoid calling fetchOrGetRemoteConfig in sendAnalyticsEvent.

Thinking out loud, could we check if configuration passed in here against latest cached value when we call sendQueuedAnalyticsEvents? 5 minutes for cache expiration is a random time we set. 
fetchOrGetRemoteConfig gets called in every client classes in functions like tokenize. 

This is all assuming that these values need to be expired during a session which I am not 100% sure about.

Reply:
I was thinking about notification pattern in each payment client. That seems like path that doesn't require a lot of code churn.

Reply:
That's effectively what happens with barrier or any kind of locking mechanism. We just need to move it to ConfigurationLoader class. And this needs to be done before `sendAnalyticsEvent` as I suggested, like in Android if we are sending values from the config into the analytics.

Reply:
> However, upon review, it appears that we only need the BTConfiguration object to send the merchantID and the FPTI environment

Some merchants do use multiple merchantIDs (for ex: Gap / Old Navy are 1 company but use different auth credentials for different merchantIDs for each sub-business)

Reply:
I understand that some merchants use different merchant IDs. My question is whether this can change at runtime. For example, is it possible for a user to be making a purchase using our SDK and the merchant ID changes from GAP to Old Navy in a subsequent request after our cache expires?